aibolit/utils/java_class.py

```diff
@@ -0,0 +1,66 @@
+# The MIT License (MIT)
+#
+# Copyright (c) 2020 Aibolit
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+
+from functools import lru_cache
+
+from typing import Iterator, TYPE_CHECKING
+from networkx import DiGraph  # type: ignore
+
+from aibolit.utils.ast import AST, ASTNodeType
+from aibolit.utils.java_class_method import JavaClassMethod
+from aibolit.utils.java_class_field import JavaClassField
+
+if TYPE_CHECKING:
+    from aibolit.utils.java_package import JavaPackage
+
+
+class JavaClass(AST):
+    def __init__(self, tree: DiGraph, root: int, java_package: 'JavaPackage'):
```

Comment:
```def __init__(self, tree: DiGraph, root: int, java_package: 'JavaPackage'):```

what's the purpose of the last argument?

Reply:
It is a reference to its owning package. So having JavaClass instance you can always obtain its JavaPackage instance, and, for example, check other classes in this package.